src/data/VariantSelector.ts

```diff
@@ -167,6 +168,49 @@ export function isValidPangoLineageQuery(query: string): boolean {
   return /^([A-Z]){1,3}(\.[0-9]{1,3})*(\.?\*)?$/.test(query.toUpperCase());
 }
 
+const stripNumber = (str: string) => Number.parseInt(str.replace(/\D+/g, ''));
+const isNumeric = (str: string) => /^\d+$/.test(str);
+
+type geneType = {
+  name: string;
+  startPosition: number;
+  endPosition: number;
+  aaSeq: string;
+};
+
+const formatGeneName = (gene: string): string => {
+  return gene.length === 1
+    ? gene.toUpperCase()
+    : gene.slice(0, -1).toUpperCase() + gene.slice(-1).toLowerCase();
+};
+
+const loadRefData = () => JSON.parse(JSON.stringify(jsonRefData));
+
+export const normalizeMutationName = (name: string) => {
+  const refData = loadRefData();
+  let items = name.split(':');
+  if (name.toLowerCase().startsWith('ins_')) {
+    return items.length === 3
+      ? `ins_${formatGeneName(items[0].substring(4))}:${items[1]}:${items[2].toUpperCase()}`
+      : `${items[0].toLowerCase()}:${items[1].toUpperCase()}`;
```

Comment:
A bit of code documentation would be useful here because it's not entirely obvious what these cases mean. Maybe just something like `// AA insertion` and `// Nuc insertion`?